src/usage/context/usage.tsx

```diff
@@ -132,6 +158,58 @@ export const UsageProvider: FC<Props> = React.memo(({children}) => {
     handleGetBillingDate()
   }, [handleGetBillingDate])
 
+  const getComputedUsage = (vector_name: string, csvData: string) => {
+    const vectors = {
+      storage_gb: v => v * 0.02,
+      writes_mb: v => v * 0.002,
+      reads_gb: v => v * 0.09,
+      query_count: v => (v * 0.01) / 100,
+    }
+
+    const values = fromFlux(csvData).table.getColumn(vector_name) as Array<any>
+    if (!values) {
+      return 0
+    }
+
+    const tot = values.reduce((a, b) => a + b, 0)
+    return vectors[vector_name](tot)
+  }
+
+  const handleGetCreditUsage = useCallback(() => {
+    try {
+      setCreditUsageStatus(RemoteDataState.Loading)
+      const vectors = ['storage_gb', 'writes_mb', 'reads_gb', 'query_count']
+      const promises = []
+
+      vectors.forEach(vector_name => {
+        promises.push(
+          getUsage({vector_name, query: {range: '30d'}}).then(resp => {
+            if (resp.status !== 200) {
+              throw new Error(resp.data.message)
+            }
+
+            return new Promise(resolve =>
+              resolve(getComputedUsage(vector_name, resp.data))
+            )
+          })
+        )
+      })
+
+      Promise.all(promises).then(result => {
+        const usage: number = result
+          .reduce((a: number, b) => a + parseFloat(b), 0)
+          .toFixed(2)
+        setCreditUsage(usage)
+        setCreditUsageStatus(RemoteDataState.Done)
```

Comment:
just thinking out loud that these could probably be coupled together to reduce a rerender cycle